Page components don't support export { default } syntax

## Description

Page components in `src/pages/` don't support `export { default } from "...";` syntax. Whenever any of my page components has this syntax, I'm getting the following build error on `npm run build`:

```
Using environment config: 'local'
success open and validate gatsby-configs - 0.088s
success load plugins - 0.390s
success onPreInit - 0.002s
success delete html and css files from previous builds - 0.017s
success initialize cache - 0.005s
success copy gatsby files - 0.028s
success onPreBootstrap - 0.009s
success createSchemaCustomization - 0.004s
success source and transform nodes - 0.017s
success building schema - 0.160s
success createPages - 0.001s

 ERROR #11328

A page component must export a React component for it to be valid. Please make sure this file exports a React component:

undefined

not finished createPagesStatefully - 0.079s
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! landing-website-gatsby@0.1.0 build: `gatsby build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the landing-website-gatsby@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/robert/.npm/_logs/2020-07-21T19_20_12_218Z-debug.log
```

As soon as I refactor `export { default } from "...";` to `export { default as default } from "...";` the build process runs without errors. But then the linter raises warnings: `warning  Export default unnecessarily renamed  no-useless-rename`.

It's also worth noting that any non-page components in `src/components/` can have `export { default } from "...";` and the build process still works fine. The problem only appears in page components in `src/pages/`.

So I assume it's some part of the building pages process that doesn't support that syntax.

Here is [documentation of export statement](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export), which clearly lists `export { deafult }` as a valid syntax.

### Steps to reproduce

1. Rename `src/pages/index.js` to `src/pages/Home.js`.
2. Create `src/pages/index.js` with the following contents:
    ```javascript
    export { default } from "./Home";
    ```
3. Run `npm run build`.
4. Change contents of `src/pages/index.js` to:
    ```javascript
    export { default as default } from "./Home";
    ```
5. Run `npm run build` again.

### Expected result

The build should run fine both times.

### Actual result

The first build ends with an error pasted below. The second build runs successfully, but the linter reports some warnings on `no-useless-rename` rule.

```
success createPages - 0.001s

 ERROR #11328

A page component must export a React component for it to be valid. Please make sure this file exports a React component:
```3

### Environment


  System:
    OS: Linux 5.7 Arch Linux
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 14.5.0 - /sbin/node
    Yarn: 1.22.4 - /sbin/yarn
    npm: 6.14.6 - /sbin/npm
  Languages:
    Python: 3.8.3 - /sbin/python

--- a/packages/gatsby/src/utils/validate-page-component.ts
+++ b/packages/gatsby/src/utils/validate-page-component.ts
@@ -81,7 +81,8 @@ export function validatePageComponent(
         fileContent.includes(`module.exports`) ||
         fileContent.includes(`exports.default`) ||
         fileContent.includes(`exports["default"]`) ||
-        fileContent.match(/export \{.* as default.*\}/s)
+        fileContent.match(/export \{.* as default.*\}/s) ||
+        fileContent.match(/export \{\s*default\s*\}/s)
 
       if (!includesDefaultExport) {
         return {

